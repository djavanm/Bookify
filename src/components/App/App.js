import React, { Component } from 'react';
import '../../styles/App.css';
import Nav from '../Nav/Nav';

class App extends Component {
  constructor() {
    super()
    this.state = {
      books: [],
      user: null
    }
  }

  render() {
    const { user } = this.state;
    return (
      <main className='app'>
        <Nav user={user} />
      </main>
    )
  }
}

export default App;
