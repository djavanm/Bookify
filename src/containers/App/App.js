import React, { Component } from 'react';
import Nav from '../../containers/Nav/Nav';
import BookContainer from '../../components/BookContainer/BookContainer';
import SearchForm from '../SearchForm/SearchForm';
import LoginForm from '../LoginForm/LoginForm';
import BookDetails from '../../components/BookDetails/BookDetails';
import { fetchOnLoad, postFavorite, deleteFavorite} from '../../util/apiCalls';
import { bindActionCreators } from 'redux';
import { setBooks, addFavorite, setFavorites, setGenres, addGenre } from '../../actions';
import { connect } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';

class App extends Component {

  componentDidMount() {
    fetchOnLoad()
      .then(data => this.props.setBooks(data))
      .catch(error => console.log(error))
  }

  toggleFavorite = (book, bool) => {
    const { currentUser, addFavorite, setFavorites, setGenres, addGenre} = this.props;
    if(!bool) {
      postFavorite(book, currentUser.id)
        .then(data => addFavorite(data))
        .then(data => addGenre(data.newFavorite.primary_genre_name))
        .catch(error => console.log(error))
    } else {
      deleteFavorite(currentUser.id, book.book_id)
        .then(data => setFavorites(data.favorites))
        .then(data => setGenres(data.foundFavorites))
        .catch(error => console.log(error))
    }
  }

  render() {
    const { currentUser } = this.props;
    return (
      <main className='app'>
      <Route exact path='/login' render={() => currentUser ? <Redirect to='/' /> : <LoginForm /> } />
      <Route exact path='/' render={() => <Nav currentUser={currentUser}/>} />
      <Route exact path='/' render={() => <SearchForm /> } />
      <Route exact path='/' render={() => <BookContainer all={true} toggleFavorite={this.toggleFavorite} /> } />
      <Route exact path='/my-collection' render={() => <Nav home={true} currentUser={currentUser}/>} />
      <Route exact path='/my-collection' render={() => <BookContainer all={false} toggleFavorite={this.toggleFavorite} /> } />
      <Route path='/book/:id' render={({ match }) => {
          let targetBook = this.props.currentBooks.find(book => book.book_id === parseInt(match.params.id));
          return <BookDetails {...targetBook} />
        }} />
      </main>
    )
  }
}

export const mapStateToProps = state => ({
  currentBooks: state.currentBooks,
  currentUser: state.currentUser,
  favorites: state.favorites
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ setBooks, addFavorite, setFavorites, setGenres, addGenre }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(App);
