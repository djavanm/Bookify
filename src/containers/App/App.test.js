import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  const currentUserMock = {
    name: 'Sam',
    email: 'blah@gmail.com',
    password: 'password'
  }
  const setBooksMock = jest.fn();
  const showStateMock = jest.fn();
  const addFavoriteMock = jest.fn();
  const setFavoritesMock = jest.fn();
  const setGenresMock = jest.fn();
  const addGenreMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App
      currentUser={currentUserMock}
      setBooks={setBooksMock}
      showState={showStateMock}
      addFavorite={addFavoriteMock}
      setFavorites={setFavoritesMock}
      setGenres={setGenresMock}
      addGenre={addGenreMock}
       />)
  })

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
