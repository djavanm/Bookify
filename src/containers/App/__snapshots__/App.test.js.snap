// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the data passed through 1`] = `
<main
  className="app"
>
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/my-collection"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/my-collection"
    render={[Function]}
  />
  <Route
    path="/book/:id"
    render={[Function]}
  />
</main>
`;
